progressReport/app.py

```diff
@@ -65,6 +66,34 @@
 
 app = Flask(__name__)
 
+"""
+Validates the fields of the mastery learning post request
+"""
+def validate_mastery_learning_post_request(request_as_json):
+    schema = {
+        "type": "object",
+        # Matches mastery fields (fields other than "school" and "class"
+        "properties": {
+            # These fields are optional.
+            "school": {"type": "string"},
+            "class": {"type": "string"},
+        },
+        "additionalProperties": {
+            "type": "object",
+            "properties": {
+                "key1": {"type": "integer"},
+                "key2": {"type": "integer"}
+            },
+            "additionalProperties": False
+        },
+    }
```

Comment:
If these values aren’t present, the code uses the following defaults:
student_mastery and class_mastery for an absent concept: 0
School: “Berkeley”
Class: “CS10”

Using default values for concepts which are not present allows us to include concepts which have not yet been added to the HAID (and are not part of the POST from GV) in the CM config file. However, I realize I made a couple of assumptions about how this would look from GV’s end; I will defer the decision of whether to use these defaults to you.